### PR TITLE
Implement openDS 0.4.0

### PR DESCRIPTION
**Pull request by samleeflang**
- Update to data model 0.4.0
- Agent changed so that it match the expectations with the new model (see AgentUtils class)
- MIDS calculation fixed so it checks the agent list instead of individual fields

- Failed to add DOI as Identifier to the object, need refactoring first. Will create a task for it

### CODE REVIEW

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java [hunk 1]
```diff
@@ -112,13 +120,32 @@ private boolean compliesToMidsTwoBio(DigitalSpecimen attributes) {
...
+        attributes.getOdsHasEvents().get(0).getDwcEventDate(),
+        attributes.getOdsHasEvents().get(0).getDwcVerbatimEventDate(),
+        convertInteger(attributes.getOdsHasEvents().get(0).getDwcYear())))
         && (eventIsPresent(attributes) &&
-        hasValue(attributes.getOdsHasEvent().get(0).getDwcFieldNumber(),
-            attributes.getOdsHasEvent().get(0).getDwcRecordNumber()))
-        && hasValue(attributes.getDwcRecordedBy(), attributes.getDwcRecordedByID());
+        hasValue(attributes.getOdsHasEvents().get(0).getDwcFieldNumber()
+//            ,
+//            attributes.getOdsHasEvents().get(0).getDwcRecordNumber())
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java:129]: Whoa will fix this, this was waiting on the discussion with Wouter on RecordNumber

Comment (samleeflang) [reply src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java]: Removed recordNumber from MIDS calculation as it is not in openDS 0.4.0

File: src/main/java/eu/dissco/core/digitalspecimenprocessor/domain/AgenRoleType.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package eu.dissco.core.digitalspecimenprocessor.domain;
+
+public enum AgenRoleType {
```
Comment (southeo) [inline src/main/java/eu/dissco/core/digitalspecimenprocessor/domain/AgenRoleType.java:3]: agenT

**Review by southeo — APPROVED** [review]
one comment not blocking
:dodo:

**Pull request merged**